src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs

```diff
@@ -0,0 +1,50 @@
+﻿using OSPSuite.Core.Domain.Services;
+
+namespace OSPSuite.Core.Domain.Builder
+{
+   public class ExpressionProfileBuildingBlock : StartValueBuildingBlock<ExpressionParameter>
+   {
+      public override string Icon => Type.IconName;
+
+      public virtual string MoleculeName { get; private set; }
+
+      public string Species { get; private set; }
+
+      public ExpressionType Type { set; get; }
+
+      public int PKSimVersion { set; get; }
+
+      public virtual string Category { get; private set; }
+
+      public override string Name
+      {
+         get => PKSimCoreConstants.ContainerName.ExpressionProfileName(MoleculeName, Species, Category);
+         set
+         {
+            if (string.Equals(Name, value))
+            {
+               return;
+            }
+
+            var (moleculeName, species, category) = PKSimCoreConstants.ContainerName.NamesFromExpressionProfileName(value);
+            if (string.IsNullOrEmpty(moleculeName))
+               return;
+
+            Species = species;
+            Category = category;
+            MoleculeName = moleculeName;
+         }
+      }
+
+      public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager cloneManager)
+      {
+         base.UpdatePropertiesFrom(source, cloneManager);
+         var sourceExpressionProfile = source as ExpressionProfileBuildingBlock;
+         if (sourceExpressionProfile == null) return;
+
+         Type = sourceExpressionProfile.Type;
+         PKSimVersion = sourceExpressionProfile.PKSimVersion;
+         Name = sourceExpressionProfile.Name;
```

Comment:
Why is this required? Isn't done by the base. method?

Reply:
Speaking about the Name here. I don't think we need this

Reply:
Yeap, it is done in ObjectBase.cs 
removed

Reply:
To use that base method name clone we would need to change how we store this stuff. We allow name to be set, but it gets decomposed and stored in the other 3 properties. When recalled, we build the name from those properties again.

If we want to use the ObjectBase name clone, then we need to reverse that storage/recall strategy to store all the properties in the _name, then decompose when recalled for Species etc.

```
      public virtual void UpdatePropertiesFrom(IUpdatable source, ICloneManager cloneManager)
      {
         var sourceObjectBase = source as IObjectBase;
         if (sourceObjectBase == null) return;

         //ID should not be updated since it should have been defined in constructor or during the object construction process
         _name = sourceObjectBase.Name;
         Icon = sourceObjectBase.Icon;
         Description = sourceObjectBase.Description;
      }
```

Reply:
or put `Name = sourceObjectBase.Name;` instead of `_name = sourceObjectBase.Name;`

Reply:
We are going to commit a change with the Name being assigne to in UpdatePropertiesFrom override - changing the objectBase seems a bit dangerous

Reply:
Let's add a comment why it's required then. THis just felt wrong
And ObjectBase should probably update Name to be honest. This looks wrong to me